datetime/src/main/scala/com/fortysevendeg/scalacheck/datetime/jdk8/granularity/LocalDateGranularity.scala

```diff
@@ -0,0 +1,42 @@
+package com.fortysevendeg.scalacheck.datetime.jdk8.granularity
+
+import java.time.LocalDate
+
+import com.fortysevendeg.scalacheck.datetime.Granularity
+
+/**
+  * Created by timpigden on 15/02/17.
+  * Copyright (c) Optrak Distribution Software Ltd, Ware 2016
+  */
+object LocalDateGranularity {
+  implicit val seconds: Granularity[LocalDate] = new Granularity[LocalDate] {
+    val normalize = (dt: LocalDate) => dt
+    val description = "Seconds"
+  }
+
+  implicit val minutes: Granularity[LocalDate] = new Granularity[LocalDate] {
+    val normalize = (dt: LocalDate) => dt
+    val description = "Minutes"
+  }
+
+  implicit val hours: Granularity[LocalDate] = new Granularity[LocalDate] {
+    val normalize = (dt: LocalDate) => dt
+    val description = "Hours"
+  }
+
+  implicit val days: Granularity[LocalDate] = new Granularity[LocalDate] {
+    val normalize = (dt: LocalDate) => dt
+    val description = "Days"
+  }
+
+  implicit val years: Granularity[LocalDate] = new Granularity[LocalDate] {
+    // Set the day of year before the hour as some days (very very rarely) start at 1am.
```

Comment:
There's a lot of copy/paste code in these blocks. This makes me wonder if there's an abstraction missing here. I'll have a think about the best way to approach this.